tslint.json

```diff
@@ -38,6 +38,16 @@
     "member-ordering": [ true, { "order": "fields-first" } ],
     "unified-signatures": true,
     "await-promise": true,
-    "no-empty-interface": false
+    "no-empty-interface": false,
+    "function-name": [
+      true, {
+        "method-regex": "^('?[a-z][\\w\\d]+'?|\\[.*\\])$",
+        "private-method-regex": "^('?[a-z][\\w\\d]+'?|\\[.*\\])$",
+        "protected-method-regex": "^('?[a-z][\\w\\d]+'?|\\[.*\\])$",
+        "static-method-regex": "^('?[A-Z_\\d]+'?|\\[.*\\])$",
+        "function-regex": "^('?[a-z][\\w\\d]+'?|\\[.*\\])$"
```

Comment:
Might want to check for no underscores in these regexes.

Reply:
That's true, will try to add this

Reply:
@enochtsang  I think for the sake of time (I know you want this PR to work off of), it would be best to open up an issue regarding this and I can get to it over Christmas? I believe I have addressed the rest of your comments